rdt-app/src/main/java/io/ona/rdt_app/presenter/RDTJsonFormFragmentPresenter.java

```diff
@@ -52,4 +66,34 @@ public void setUpToolBar() {
         super.setUpToolBar();
         getView().updateVisibilityOfNextAndSave(false, false);
     }
+
+    @Override
+    public void saveForm() throws JSONException {
+        interactor.saveForm(new JSONObject(getView().getCurrentJsonState()));
+    }
+
+    @Override
+    public void performNextButtonAction(String currentStep, Object isSubmit) {
+        String currStepStr = "step" + currentStep;
+        if ("step8".equals(currStepStr)) {
+            String rdtType = formFragment.getRDTType();
+            if (Constants.CARESTART_RDT.equals(rdtType)) {
+                JsonFormFragment nextFragment = RDTJsonFormFragment.getFormFragment("step14");
+                formFragment.transactFragment(nextFragment);
+            } else {
+                formFragment.moveToNextStep();
+            }
+        } else if ("step5".equals(currStepStr)) {
+            try {
+                saveForm();
+                formFragment.moveToNextStep();
```

Comment:
Is this correct? saving and then moving to next step

Reply:
Yes it is, they wanted to save an expired rdt's information before going back to the first step to re-do the test with a different rdt.